# Исследование надежности заемщиков


### Цель проекта
На основе статистики о платёжеспособности клиентов банка провести исследование оценивающее важность факторов на возврат кредита в срок.

### Цель исследования — проверить гипотезы:
1. Есть зависимость между количеством детей и возвратом кредита в срок
2. Есть зависимость между семейным положением и возвратом кредита в срок
3. Есть зависимость между уровнем дохода и возвратом кредита в срок
4. Есть зависимость между целью кредита влияют на его возврат в срок
5. Есть зависимость между возврастом заемщика и возврата кредита в срок

### Ход исследования
исследование пройдёт в несколько этапов:
 1. [Обзор данных](#обзор-данных)
 2. [Предобработка данных](#предобработка-данных)
 3. [Исследование данных и проверка гипотез](#исследование-данных-и-проверка-гипотез)
 4. [Подведение итогов и написание выводов](#общий-вывод)

## Обзор данных

Описание данных

- `children` — количество детей в семье
- `days_employed` — общий трудовой стаж в днях
- `dob_years` — возраст клиента в годах
- `education` — уровень образования клиента
- `education_id` — идентификатор уровня образования
- `family_status` — семейное положение
- `family_status_id` — идентификатор семейного положения
- `gender` — пол клиента
- `income_type` — тип занятости
- `debt` — имел ли задолженность по возврату кредитов
- `total_income` — ежемесячный доход
- `purpose` — цель получения кредита

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('C:\\Users\\Freo\\Desktop\\projects\\datasets\\p1.csv')

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

In [7]:
data['days_employed'] = data['days_employed'].abs()

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

In [15]:
data['education'] = data['education'].str.lower()

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

На основании диапазонов, указанных ниже, создадим в датафрейме `data` столбец `total_income_category` с категориями:

- 0–30000 — "E";
- 30001–50000 — "D";
- 50001–200000 — "C";
- 200001–1000000 — "B";
- 1000001 и выше — "A".

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследование данных и проверка гипотез

### 3.1 Есть зависимость между количеством детей и возвратом кредита в срок

In [23]:
display(data.head())

print(f'\n Значения в столбце debt: {data["debt"].unique()}') #проверим на наличие ошибочных значений
#все в порядке, в столбце только '1' и '0', что соответственно  скорее всего означает наличие или отстутсвие задолженности

print(f'\n Количество детей у заемщика; количество заемщиков: \n{data["children"].value_counts()}') # посмотрим какие значения есть в столбце дети, оценим достоверность выборки
#выборка не очень большая, особенно для заемщиков с 3,4,5 детьми

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы


 Значения в столбце debt: [0 1]

 Количество детей у заемщика; количество заемщиков: 
children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: count, dtype: int64


In [24]:
data_pivot_child_debt = data.pivot_table(index='children', values='debt', aggfunc=['count', 'sum','mean'])
display(data_pivot_child_debt)

count   sum      mean
           debt  debt      debt
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542
3           330    27  0.081818
4            41     4  0.097561
5             9     0  0.000000

* За исключением заемщиков с 5 детьми(где выборка совсем мала) данные получились весьма похожи,
 судя по всему наличие детей повышает риск невозврата в срок
 поскольку значения выглядят близкими, сделаем дополнительный подсчет для сравнения
 бездетных заемщиков и заемщиков с детьми

In [25]:
data['kids_yes'] = data['children'] != 0


In [26]:
data_pivot_kids_y_n_debt = data.pivot_table(index='kids_yes', values='debt', aggfunc=['count', 'sum','mean'])
data_pivot_kids_y_n_debt = data_pivot_kids_y_n_debt.reset_index()

display(data_pivot_kids_y_n_debt)

print(f'Вероятность что заемщик с детьми не заплатит вовремя выше в ~ {(data_pivot_kids_y_n_debt.loc[1][3] / data_pivot_kids_y_n_debt.loc[0][3]):.4f} раз')
#вероятность что заемщик имеющий детей не сделает платеж вовремя выше на ~22.5% по сравнению с бездетным

kids_yes  count   sum      mean
             debt  debt      debt
0    False  14091  1063  0.075438
1     True   7240   669  0.092403

Вероятность что заемщик с детьми не заплатит вовремя выше в ~ 1.2249 раз


C:\Users\Freo\AppData\Local\Temp\ipykernel_19588\703367820.py:6: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  print(f'Вероятность что заемщик с детьми не заплатит вовремя выше в ~ {(data_pivot_kids_y_n_debt.loc[1][3] / data_pivot_kids_y_n_debt.loc[0][3]):.4f} раз')


**Вывод:**
мы проверили есть ли зависимость между количеством детей и возвратом кредита в срок, и обнаружили что она действительно существует. Важно сделать оговорку что с увеличением количества детей - выборка становится все менее репрезентативной, и даже единичные случаи будут сильно смещать %, поэтому построить точную предсказательную модель для каждого конкретного значения детей не получится; однако судя по всему на имеющихся данных для всех заемщиков с детьми ожидаемый % неплательщиков примерно одинаков, и он выше чем % неплательщиков без детей. Поэтому хорошей идеей будет сравнить общий процент неплательщиков без детей и неплатещиков с детьми, тем самым увеличив выборку и сгладив погрешность.

**Есть ли зависимость между количеством детей и возвратом кредита в срок?** - есть;

Вероятность что заемщик имеющий детей не сделает платеж вовремя выше на ~22.5% по сравнению с бездетным

### 3.2 Есть зависимость между семейным положением и возвратом кредита в срок


In [27]:
# по аналогии с поиском зависимости из прошлого задания построим сводную таблицу для family_status

data_pivot_family_status_id_debt = data.pivot_table(index='family_status', values='debt', aggfunc=['count', 'sum','mean'])
display(data_pivot_family_status_id_debt)



count  sum      mean
                        debt debt      debt
family_status                              
Не женат / не замужем   2796  273  0.097639
в разводе               1189   84  0.070648
вдовец / вдова           951   63  0.066246
гражданский брак        4134  385  0.093130
женат / замужем        12261  927  0.075606

Итоговое соотношение семейного положения и возврата в срок: женат / замужем = 7.56% гражданский брак = 9.31% вдовец / вдова = 6.62% в разводе = 7.06% Не женат / не замужем = 9.76%

Данные опять разделились на две группы, с меньшим и большим риском невозврата; конечно они не полностью однородны чтобы уверенно обьяединять и оперировать только ими двумя, но различие внутри этих групп куда меньше различия "снаружи", будет интересно сравнить насколько отличаются в среднем.

In [28]:
# сначала проверим на соответствие 'family_status_id' и 'family_status' чтобы далее пользоваться ими
display(data[data['family_status_id']==0]['family_status'].head(1))
display(data[data['family_status_id']==1]['family_status'].head(1))
display(data[data['family_status_id']==2]['family_status'].head(1))
display(data[data['family_status_id']==3]['family_status'].head(1))
display(data[data['family_status_id']==4]['family_status'].head(1))
# 0 - женат / замужем; 1 - гражданский брак; 2 - вдовец / вдова; 3 - в разводе; 4 - Не женат / не замужем


0    женат / замужем
Name: family_status, dtype: object

4    гражданский брак
Name: family_status, dtype: object

18    вдовец / вдова
Name: family_status, dtype: object

19    в разводе
Name: family_status, dtype: object

24    Не женат / не замужем
Name: family_status, dtype: object

In [29]:
#зададим условия формирования групп: высокий риск "гражданский брак" и "Не женат / не замужем"
data['family_group_high_risk'] = (data['family_status_id'] == 1) | ( data['family_status_id'] ==4)

data_pivot_family_risk_cat_debt = data.pivot_table(index='family_group_high_risk', values='debt', aggfunc=['count', 'sum','mean'])
data_pivot_family_risk_cat_debt=data_pivot_family_risk_cat_debt.reset_index()
display(data_pivot_family_risk_cat_debt)
print(f'Вероятность что заемщик категории высокого риска не заплатит вовремя выше в ~'
      f'{(data_pivot_family_risk_cat_debt.loc[1][3] / data_pivot_family_risk_cat_debt.loc[0][3]):.4f} раз')



family_group_high_risk  count   sum      mean
                           debt  debt      debt
0                  False  14401  1074  0.074578
1                   True   6930   658  0.094949

Вероятность что заемщик категории высокого риска не заплатит вовремя выше в ~1.2732 раз


C:\Users\Freo\AppData\Local\Temp\ipykernel_19588\4048275479.py:8: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  f'{(data_pivot_family_risk_cat_debt.loc[1][3] / data_pivot_family_risk_cat_debt.loc[0][3]):.4f} раз')


**Вывод:** 
несмотря на не очень большой размер выборки в представленных данных удалось найти определенные закономерности:
самый низкий процент невозврата у катергии "вдовец\вдова", затем 'в разводе' и 'женат \ замужем'.
С учетом того что выборка для "вдовец \ вдова" и "в разводе" весьма небольшая, дополнительные случаи невозврата могут слегка изменить позиции этих трех категорий относительно друг друга.

Процент невозврата вовремя у категорий "гражданский брак" и "Не женат / не замужем" существенно выше (9.31% и 9.76% соотв.) что позволяет сделать предположение о прямой зависимости между возврастом; те кто находящится в гражданском браке и не женатые (не замужние) как правило - молоды, а точнее вероятность этого, скорее всего, обратно пропорциональна возрасту



**Есть ли зависимость между семейным положением и возвратом кредита в срок?** - есть;

Процент невозврата у катергии "вдовец\вдова", 'в разводе', 'женат \ замужем' = ниже, а у категорий "гражданский брак", "не женат \ не замужем" = выше, в среднем примерно на 27%



### 3.3 Есть зависимость между уровнем дохода и возвратом кредита в срок

In [30]:
# по аналогии с прошлым заданием построим сводную таблицу
data_pivot_income_cat_debt = data.pivot_table(index='total_income_category', values='debt', aggfunc=['count', 'sum','mean'])
data_pivot_income_cat_debt = data_pivot_income_cat_debt.reset_index()
display(data_pivot_income_cat_debt)


total_income_category  count   sum      mean
                          debt  debt      debt
0                     A     25     2  0.080000
1                     B   5014   354  0.070602
2                     C  15921  1353  0.084982
3                     D    349    21  0.060172
4                     E     22     2  0.090909

категории для разграничения уровней дохода мы определили ранее;

после анализа статистики видно что категории "A" и "E" практически не пользуются услугами кредитования, (при этом даже высокий доход совсем не гарантирует что оплата будет вовремя, несмотря на то что выборка по доходу до 30000 и более 1м совсем мала и непрезентативна, мы получили забавную ситуацию с практически идентичным % не заплативших вовремя)

видим что подавляющее большинство заемщиков из полученных данных попали в категорию "С", возможно следовало выбирать категории на этом отрезке меньшим шагом, к примеру 50-100к, 100-150, 150-200  \ начав ее "выше", подняв засчет этого верхнюю границу "D" чтобы получить больше возможностей для сравнения.

видим что в категории "D" (несмотретря на небольшую выборку) процент задержки выплат ниже чем в категориях с самым высоким доходом

видим что при росте зарплат при сравнении "C" и "B" количество задержек снижается

In [31]:
print(
    f'При сравнении категорий "C" и "B" вероятность что заемщик из категоии "С" не заплатит вовремя выше в ~ '
      f'{(data_pivot_income_cat_debt.loc[2][3] / data_pivot_income_cat_debt.loc[1][3]):.4f} раз')

При сравнении категорий "C" и "B" вероятность что заемщик из категоии "С" не заплатит вовремя выше в ~ 1.2037 раз


C:\Users\Freo\AppData\Local\Temp\ipykernel_19588\894634680.py:3: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  f'{(data_pivot_income_cat_debt.loc[2][3] / data_pivot_income_cat_debt.loc[1][3]):.4f} раз')


**Вывод:** 
**Есть ли зависимость между уровнем дохода и возвратом кредита в срок?** - есть, но:!

Судя по всему можно сделать выводы только для сравнения высокого и очень высокого уровня дохода:

Заемщики с доходом 50001–200000 задерживают платежи по кредиту в среднем на 20% чаще чем замщики с доходом 200001–1000000.

чтобы выявить зависимость для среднего и высокого уровня доходов необходимо изменить границы категорий

### 3.4 Есть зависимость между целью кредита влияют на его возврат в срок

In [32]:
# по аналогии с прошлым заданием построим сводную таблицу
data_pivot_purpose_cat_debt = data.pivot_table(index='purpose_category', values='debt', aggfunc=['count', 'sum','mean'])

data_pivot_purpose_cat_debt = data_pivot_purpose_cat_debt.reset_index()
print(data_pivot_purpose_cat_debt)

           purpose_category  count  sum      mean
                              debt debt      debt
0    операции с автомобилем   4279  400  0.093480
1  операции с недвижимостью  10751  780  0.072551
2     получение образования   3988  369  0.092528
3        проведение свадьбы   2313  183  0.079118


* Посчитаем по аналогии сравнение для категорий "высокого" и "низкого" рисков невозврата, сначала включая "свадьбу" в категории низкого риска, а затем относя ее к средним - и сравнивая только категории авто+образование с кредитами категории недвижимость

In [33]:
data['purpose_high_risk'] = (data['purpose_category'] == 'получение образования') | ( data['purpose_category'] =='операции с автомобилем')

data_pivot_purpose_cat_high_risk_debt = data.pivot_table(index='purpose_high_risk', values='debt', aggfunc=['count', 'sum','mean'])
data_pivot_purpose_cat_high_risk_debt = data_pivot_purpose_cat_high_risk_debt.reset_index()

display(data_pivot_purpose_cat_high_risk_debt)

print(
    f'Вероятность невыплаты в группе высокого риска выше в '
      f'{(data_pivot_purpose_cat_high_risk_debt.loc[1][3] / data_pivot_purpose_cat_high_risk_debt.loc[0][3]):.4f}'
    f' раз чем в группе низкого'
)

print(f'Вероятность невыплаты в группе высокого риска выше в '
    f'{data_pivot_purpose_cat_high_risk_debt.loc[1][3]/data_pivot_purpose_cat_debt.loc[1][3]:.4f} раз '
      f'чем вероятность невыплаты в категории недвижимости'
     ) 

purpose_high_risk  count  sum      mean
                      debt debt      debt
0             False  13064  963  0.073714
1              True   8267  769  0.093020

Вероятность невыплаты в группе высокого риска выше в 1.2619 раз чем в группе низкого
Вероятность невыплаты в группе высокого риска выше в 1.2821 раз чем вероятность невыплаты в категории недвижимости


C:\Users\Freo\AppData\Local\Temp\ipykernel_19588\1313574027.py:10: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  f'{(data_pivot_purpose_cat_high_risk_debt.loc[1][3] / data_pivot_purpose_cat_high_risk_debt.loc[0][3]):.4f}'
C:\Users\Freo\AppData\Local\Temp\ipykernel_19588\1313574027.py:15: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  f'{data_pivot_purpose_cat_high_risk_debt.loc[1][3]/data_pivot_purpose_cat_debt.loc[1][3]:.4f} раз '


* наименьший процент задержек по выплатам ( \~7.26%) в категории "операции с недвижимостью", включающей в себя ремонт и покупку жилья(при этом эта категория является самой распространенной);

* затем, "на проведение свадьбы" ( \~7.91% задержек);

* затем 'получение образования' и 'операции с автомобилем' (\~9.25% и \~9.35% соотв.)

Разница между включением и не включением кредитов на свадьбу в группу низкого риска выглядит небольшой, но это из-за разного веса в выборке, размеры выборок отличаются почти в 5 раз, и кредиты на недвижимость "размывают" дельту засчет своей многочисленности, поэтому здесь я бы не торопился обьединять их в одно, особенно с учетого разной природы их целеполагания.

> _Разница в задержках по выплатам в категории недвижимости в сравнении с образованием и автомобилем почти 28%!_




**Вывод:** 


**Как разные цели кредита влияют на его возврат в срок?**
кредиты взятые в категории "операции с недвижимостью"существенно повышают вероятность их возврата в срок, можно предположить что это связанно с осознанным заранее запланированным решением о кредите;
а так же можно предположить что у категорий "получение образования" и "операции с автомобилем" есть зависимость с возрастом; те кто берут кредит на образование - как правило - молоды.



### 3.5 Есть зависимость между возрастом заемщика и возврата кредита в срок

In [34]:
def categorize_age(age):
    try:
        if 0 <= age <= 30:
            return 'E'
        elif 31 <= age <= 40:
            return 'D'
        elif 41 <= age <= 50:
            return 'C'
        elif 51 <= age <= 60:
            return 'B'
        elif age >= 61:
            return 'A'
    except:
        pass


data['age_category'] = data['dob_years'].apply(categorize_age)

data_pivot_age_debt = data.pivot_table(index='age_category', values='debt', aggfunc=['count', 'sum', 'mean'])

display(data_pivot_age_debt)

count  sum      mean
              debt debt      debt
age_category                     
A             2118  101  0.047686
B             4496  279  0.062055
C             5225  400  0.076555
D             5693  541  0.095029
E             3799  411  0.108186

гипотеза подтвердилась, вероятность возврата кредита вовремя обратно пропорциальна возрасту 

### Общий вывод

После анализа даннных можно сделать следующие выводы:

- Вероятность что заемщик имеющий детей не сделает платеж вовремя выше на ~22.5% по сравнению с бездетным

- вероятность что заемщик в статусе "не замужем \ не женат" и "гражданский брак не сделает платеж вовремя выше на ~27% по сравнению с другими категориями

- при сравнении высокого (50001–200000) и очень высокого уровня (200001–1000000) вероятность что заемщик с высоким доходом не сделает платеж вовремя выше на ~20% . Чтобы выявить зависимость (или удостоверится в ее отстуствии) для среднего и высокого уровня доходов необходимо изменить границы категорий

- вероятность что заемщик взявший в категории "операции с автомобилем" и "получение образования" не сделает платеж вовремя выше на ~26% чем заемщик в категории "операции с недвижимостью"



- самая "надежная" часть диапазона заемщиков = без детей + для недвижимости + в браке(изза разамеров выборки)

- так же подтвердилась гипотеза что есть прямая зависимость возраста заемщика и вероятностью платежа вовремя (чем старше тем больше)